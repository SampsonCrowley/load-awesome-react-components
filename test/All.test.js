import React from 'react';
import Loaders from '../build/rrpm.module';
import renderer from 'react-test-renderer';

const loaderKeys = Object.keys(Loaders)

for(let l = 0; l < loaderKeys.length; l++){
  const sectionName = loaderKeys[l],
        section = Loaders[sectionName],
        sectionKeys = Object.keys(section)

  for(let i = 0; i < sectionKeys.length; i++){
    const name = sectionKeys[i],
          Component = section[name]
    describe(`${sectionName}::${name}`, () => {
      const component = renderer.create(
        <Component className='test' style={{height: '100px', width: '100px'}} />
      );

      let tree;

      it('should render correctly', () => {
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });

      it('should accept additional classNames', () => {
        expect(tree.props.className).toMatch(/\s+test$/)
      })

      it('should accept all other standard props', () => {
        expect(tree.props.style).toBeTruthy()
        expect(tree.props.style.height).toBe('100px')
        expect(tree.props.style.width).toBe('100px')
      })

    });
  }
}
