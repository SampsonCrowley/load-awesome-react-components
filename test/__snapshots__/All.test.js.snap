// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ball::Atom should render correctly 1`] = `
<div
  className="la-ball-atom test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
  <div />
  <div />
  <div />
</div>
`;

exports[`Ball::Beat should render correctly 1`] = `
<div
  className="la-ball-beat test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
  <div />
  <div />
</div>
`;

exports[`Ball::Circus should render correctly 1`] = `
<div
  className="la-ball-circus test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
  <div />
  <div />
  <div />
  <div />
</div>
`;

exports[`Ball::ClimbingDot should render correctly 1`] = `
<div
  className="la-ball-climbing-dot test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
  <div />
  <div />
  <div />
</div>
`;

exports[`Ball::ClipRotate should render correctly 1`] = `
<div
  className="la-ball-clip-rotate test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
</div>
`;

exports[`Ball::ClipRotateMultiple should render correctly 1`] = `
<div
  className="la-ball-clip-rotate-multiple test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
  <div />
</div>
`;

exports[`Ball::ClipRotatePulse should render correctly 1`] = `
<div
  className="la-ball-clip-rotate-pulse test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
  <div />
</div>
`;

exports[`Ball::EightBits should render correctly 1`] = `
<div
  className="la-ball-8bits test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
</div>
`;

exports[`Ball::ElasticDots should render correctly 1`] = `
<div
  className="la-ball-elastic-dots test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
  <div />
  <div />
  <div />
  <div />
</div>
`;

exports[`Ball::Fall should render correctly 1`] = `
<div
  className="la-ball-fall test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
  <div />
  <div />
</div>
`;

exports[`Ball::Fusion should render correctly 1`] = `
<div
  className="la-ball-fusion test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
  <div />
  <div />
  <div />
</div>
`;

exports[`Ball::GridBeat should render correctly 1`] = `
<div
  className="la-ball-grid-beat test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
</div>
`;

exports[`Ball::GridPulse should render correctly 1`] = `
<div
  className="la-ball-grid-pulse test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
</div>
`;

exports[`Ball::NewtonCradle should render correctly 1`] = `
<div
  className="la-ball-newton-cradle test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
  <div />
  <div />
  <div />
</div>
`;

exports[`Ball::Pulse should render correctly 1`] = `
<div
  className="la-ball-pulse test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
  <div />
  <div />
</div>
`;

exports[`Ball::PulseRise should render correctly 1`] = `
<div
  className="la-ball-pulse-rise test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
  <div />
  <div />
  <div />
  <div />
</div>
`;

exports[`Ball::PulseSync should render correctly 1`] = `
<div
  className="la-ball-pulse-sync test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
  <div />
  <div />
</div>
`;

exports[`Ball::Rotate should render correctly 1`] = `
<div
  className="la-ball-rotate test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
</div>
`;

exports[`Ball::RunningDots should render correctly 1`] = `
<div
  className="la-ball-running-dots test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
  <div />
  <div />
  <div />
  <div />
</div>
`;

exports[`Ball::Scale should render correctly 1`] = `
<div
  className="la-ball-scale test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
</div>
`;

exports[`Ball::ScaleMultiple should render correctly 1`] = `
<div
  className="la-ball-scale test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
</div>
`;

exports[`Ball::ScalePulse should render correctly 1`] = `
<div
  className="la-ball-scale-pulse test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
  <div />
</div>
`;

exports[`Ball::ScaleRipple should render correctly 1`] = `
<div
  className="la-ball-scale-ripple test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
</div>
`;

exports[`Ball::ScaleRippleMultiple should render correctly 1`] = `
<div
  className="la-ball-scale-ripple-multiple test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
  <div />
  <div />
</div>
`;

exports[`Ball::Spin should render correctly 1`] = `
<div
  className="la-ball-spin test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
</div>
`;

exports[`Ball::SpinClockwise should render correctly 1`] = `
<div
  className="la-ball-spin-clockwise test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
</div>
`;

exports[`Ball::SpinClockwiseFade should render correctly 1`] = `
<div
  className="la-ball-spin-clockwise-fade test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
</div>
`;

exports[`Ball::SpinClockwiseFadeRotating should render correctly 1`] = `
<div
  className="la-ball-spin-clockwise-fade-rotating test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
</div>
`;

exports[`Ball::SpinFade should render correctly 1`] = `
<div
  className="la-ball-spin-fade test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
</div>
`;

exports[`Ball::SpinFadeRotating should render correctly 1`] = `
<div
  className="la-ball-spin-fade-rotating test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
</div>
`;

exports[`Ball::SpinRotate should render correctly 1`] = `
<div
  className="la-ball-spin-rotate test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
  <div />
</div>
`;

exports[`Ball::SquareClockwiseSpin should render correctly 1`] = `
<div
  className="la-ball-square-clockwise-spin test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
</div>
`;

exports[`Ball::SquareSpin should render correctly 1`] = `
<div
  className="la-ball-square-spin test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
</div>
`;

exports[`Ball::TrianglePath should render correctly 1`] = `
<div
  className="la-ball-triangle-path test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
  <div />
  <div />
</div>
`;

exports[`Ball::ZigZag should render correctly 1`] = `
<div
  className="la-ball-zig-zag test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
  <div />
</div>
`;

exports[`Ball::ZigZagDeflect should render correctly 1`] = `
<div
  className="la-ball-zig-zag-deflect test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
  <div />
</div>
`;

exports[`Line::Scale should render correctly 1`] = `
<div
  className="la-line-scale test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
  <div />
  <div />
  <div />
  <div />
</div>
`;

exports[`Line::ScaleParty should render correctly 1`] = `
<div
  className="la-line-scale-party test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
  <div />
  <div />
  <div />
  <div />
</div>
`;

exports[`Line::ScalePulseOut should render correctly 1`] = `
<div
  className="la-line-scale-pulse-out test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
  <div />
  <div />
  <div />
  <div />
</div>
`;

exports[`Line::ScalePulseOutRapid should render correctly 1`] = `
<div
  className="la-line-scale-pulse-out-rapid test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
  <div />
  <div />
  <div />
  <div />
</div>
`;

exports[`Line::SpinClockwiseFade should render correctly 1`] = `
<div
  className="la-line-spin-clockwise-fade test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
</div>
`;

exports[`Line::SpinClockwiseFadeRotating should render correctly 1`] = `
<div
  className="la-line-spin-clockwise-fade-rotating test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
</div>
`;

exports[`Line::SpinFade should render correctly 1`] = `
<div
  className="la-line-spin-fade test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
</div>
`;

exports[`Line::SpinFadeRotating should render correctly 1`] = `
<div
  className="la-line-spin-fade-rotating test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
</div>
`;

exports[`Misc::Cog should render correctly 1`] = `
<div
  className="la-cog test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
</div>
`;

exports[`Misc::CubeTransition should render correctly 1`] = `
<div
  className="la-cube-transition test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
  <div />
</div>
`;

exports[`Misc::Fire should render correctly 1`] = `
<div
  className="la-fire test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
  <div />
  <div />
</div>
`;

exports[`Misc::Pacman should render correctly 1`] = `
<div
  className="la-pacman test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
</div>
`;

exports[`Misc::Timer should render correctly 1`] = `
<div
  className="la-timer test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
</div>
`;

exports[`Square::JellyBox should render correctly 1`] = `
<div
  className="la-square-jelly-box test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
  <div />
</div>
`;

exports[`Square::Loader should render correctly 1`] = `
<div
  className="la-square-loader test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
</div>
`;

exports[`Square::Spin should render correctly 1`] = `
<div
  className="la-square-spin test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
</div>
`;

exports[`Triangle::SkewSpin should render correctly 1`] = `
<div
  className="la-triangle-skew-spin test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div />
</div>
`;
